engine-runner-bin/build.rs

```diff
@@ -0,0 +1,68 @@
+use std::{fs, path::Path};
+
+use engine_upgrade_utils::{NEW_VERSION, OLD_VERSION};
+
+fn main() {
+	// === Ensure the runner runs the linker checks at compile time ===
+
+	let out_dir = std::env::var("OUT_DIR").unwrap();
+
+	let build_dir = std::path::Path::new(&out_dir)
+		.parent()
+		.unwrap()
+		.parent()
+		.unwrap()
+		.parent()
+		.unwrap(); // target/debug or target/release
+
+	// ./old-engine-dylib from project root.
+	let old_version = build_dir.parent().unwrap().parent().unwrap().join("old-engine-dylib");
+
+	let old_version_str = old_version.to_str().unwrap();
+
+	let build_dir_str = build_dir.to_str().unwrap(); // target/debug or target/release
+
+	println!("cargo:rustc-link-search=native={old_version_str}");
+	println!("cargo:rustc-link-search=native={build_dir_str}");
+
+	let old_version_suffix = OLD_VERSION.replace('.', "_");
+	let new_version_suffix = NEW_VERSION.replace('.', "_");
+
+	println!("cargo:rustc-link-lib=dylib=chainflip_engine_v{}", old_version_suffix);
```

Comment:
We could define a constant for the lib in the utils crate: `const OLD_LIB_NAME: &str = concat!("chainflip_engine_v", OLD_VERSION)`.

Then we don't have to repeat the the replace(..) boilerplate all over the place.

Reply:
Yeah not the biggest fan of this repetition either. Unfortunately .replace() is not a const fn, and I think it makes more sense to have the version specified as the standard "1.3.1" than "1_3_1".